src/test/java/org/opensearch/knn/index/LuceneEngineIT.java

```diff
@@ -449,4 +516,37 @@ private void validateQueryResultsWithFilters(
                 .containsAll(expectedDocIdsKLimitsFilterResult)
         );
     }
+
+    private void validateRadiusSearchResults(
+        final float[] searchVector,
+        final float radius,
+        final SpaceType spaceType,
+        final int expectedResults,
+        @Nullable final String filterField,
+        @Nullable final String filterValue
+    ) throws Exception {
+        KNNQueryBuilder queryBuilder;
+        if (filterField != null && filterValue != null) {
+            queryBuilder = new KNNQueryBuilder(FIELD_NAME, searchVector, radius, QueryBuilders.termQuery(filterField, filterValue));
```

Comment:
note - this is where query build pattern ends up helping simplify code

Reply:
Makes sense